### PR TITLE
Fix PlayStation serial code region parsing

### PR DESCRIPTION
**Pull request by Deterous**
- Do not assume S_J_ and S_P_ serials are Japanese
- Fix typo: PABX -> PBPX
- PCXC are Japanese
- PUBX are USA

### CODE REVIEW

**Comment by mnadareski** [comment]
Please note that this is used by both PS1 and PS2. Don't remove things or change existing cases unnecessarily.

File: MPF.Modules/BaseParameters.cs [hunk 1]
```diff
@@ -1849,9 +1849,8 @@ protected static string GetPlayStation5Version(char? driveLetter)
                     case 'A': return Region.Asia;
                     case 'C': return Region.China;
                     case 'E': return Region.Europe;
-                    case 'J': return Region.JapanKorea;
```
Comment (mnadareski) [inline MPF.Modules/BaseParameters.cs:1852]: I disagree with this change. Users are more than capable of selecting the region that their particular disc is from. There's a reason it's both JP/KR as it is now.

File: MPF.Modules/BaseParameters.cs [hunk 2]
```diff
@@ -1849,9 +1849,8 @@ protected static string GetPlayStation5Version(char? driveLetter)
...
                     case 'K': return Region.SouthKorea;
-                    case 'P': return Region.Japan;
+                    case 'P': return null; // P-region discs may be Japan, Asia or Korea
```
Comment (mnadareski) [inline MPF.Modules/BaseParameters.cs:1853]: It sounds like a more accurate thing would be to return `Region.Asia` instead, then.

File: MPF.Modules/BaseParameters.cs [hunk 3]
```diff
@@ -1861,13 +1860,17 @@ protected static string GetPlayStation5Version(char? driveLetter)
                 return Region.Japan;
 
             // Region appears entirely random
-            else if (serial.StartsWith("PABX"))
+            else if (serial.StartsWith("PBPX"))
```
Comment (mnadareski) [inline MPF.Modules/BaseParameters.cs:1863]: Keep both PABX and PBPX please.

File: MPF.Modules/BaseParameters.cs (hunk 2 above)
Comment (mnadareski) [reply MPF.Modules/BaseParameters.cs]: Thank you. Solving issues exactly as requested is not always the right course of action :)

Comment (sadikyo) [reply MPF.Modules/BaseParameters.cs]: Even if it's not always correct, I'd rather SLPM return Japan, then Asia.  It is much more common to be Japan.

Now, if you want to get a bit more specific, I could list out the number ranges that are likely to be certain regions.  But if you don't want to deal with that, that's ok.

File: MPF.Modules/BaseParameters.cs [hunk 4]
```diff
@@ -1849,10 +1849,33 @@ protected static string GetPlayStation5Version(char? driveLetter)
...
+                                }
+                            case 'M':
+                                // Check first three digits of S_PM serial
+                                switch (serial.Substring(4, 3))
+                                {
+                                    case "645": return Region.SouthKorea;
+                                    case "675": return Region.SouthKorea;
+                                    case "885": return Region.SouthKorea;
+                                    default: return Region.Japan; // Remaining S_PM serials may be Japan or Asia
+                                }
```
Comment (mnadareski) [inline MPF.Modules/BaseParameters.cs:1876]: This feels really difficult to keep up to date, I'll be honest. I understand that it's far more accurate, though.

**Pull request merged**